subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1157,9 +1148,25 @@ uint8_t ll_adv_enable(uint8_t enable)
 		conn->pause_rx_data = 0U;
 #endif /* CONFIG_BT_CTLR_LE_ENC */
 
+#if defined(CONFIG_BT_CTLR_DATA_LENGTH)
+		uint8_t phy_in_use;
+
+		if (false) {
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+		} else if (pdu_adv->type == PDU_ADV_TYPE_EXT_IND) {
+			phy_in_use = lll->phy_s;
+#else /* !CONFIG_BT_CTLR_ADV_EXT */
+		} else {
+			phy_in_use = PHY_1M;
+#endif /* !CONFIG_BT_CTLR_ADV_EXT */
+		}
+
+		ull_dle_init(conn, phy_in_use);
```

Comment:
Why it is called only in case refactored LLCPs are enabled? Isn't it not needed for legacy LLCPs?

Reply:
For legacy it is done in lines 981-994

Reply:
Ok, this is why I didn't spot it, because it is not a function.